ci/scripts/python_sdist_test.sh

```diff
@@ -42,10 +42,16 @@ export PYARROW_WITH_DATASET=${ARROW_DATASET:-OFF}
 # unset ARROW_HOME
 # apt purge -y pkg-config
 
+# ARROW-12619
+if command -v git &> /dev/null; then
+  echo "Git exists, remove it from PATH before executing this script."
+  exit 1
+fi
+
 if [ -n "${PYARROW_VERSION:-}" ]; then
   sdist="${arrow_dir}/python/dist/pyarrow-${PYARROW_VERSION}.tar.gz"
 else
-  sdist=$(ls "${arrow_dir}/python/dist/pyarrow-*.tar.gz" | sort -r | head -n1)
+  sdist=$(ls ${arrow_dir}/python/dist/pyarrow-*.tar.gz | sort -r | head -n1)
```

Comment:
very minor nit: you could keep the quoting around just `${arrow_dir}`

Reply:
Interestingly it didn't pick up the tarball with the quotes, which I wouldn't have expected either.

Reply:
```console
Creating arrow_ubuntu-python-sdist-test_run ... done
root@202aa2bdd8d0:/# ls "/arrow/python/dist/pyarrow-*.tar.gz"
ls: cannot access '/arrow/python/dist/pyarrow-*.tar.gz': No such file or directory
root@202aa2bdd8d0:/# ls /arrow/python/dist/pyarrow-*.tar.gz
/arrow/python/dist/pyarrow-5.0.0.dev866+g49641009e.d20210517.tar.gz
root@202aa2bdd8d0:/#
```

Reply:
That is expected, but `ls "${arrow_dir}"/python/dist/pyarrow-*.tar.gz` doesn't work? e.g.

```
$ export arrow_dir=/tmp/foo
$ touch /tmp/foo/pyarrow-5.0.0.tar.gz
$ ls "${arrow_dir}"/pyarrow-*.tar.gz
/tmp/foo/pyarrow-5.0.0.tar.gz
```

but either way this is a very minor nit, just if you wanted to be extra-defensive.

Reply:
`$arrow_dir` is going to be `/arrow` :)